presentation/src/main/java/team/jsv/icec/ui/main/MosaicResultActivity.kt

```diff
@@ -0,0 +1,118 @@
+package team.jsv.icec.ui.main
+
+import android.content.Intent
+import android.graphics.Color
+import android.graphics.drawable.GradientDrawable
+import android.net.Uri
+import android.os.Build
+import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import androidx.activity.viewModels
+import androidx.core.content.ContextCompat
+import com.google.android.material.snackbar.Snackbar
+import team.jsv.icec.base.BaseActivity
+import team.jsv.icec.util.gone
+import team.jsv.icec.util.visible
+import team.jsv.presentation.R
+import team.jsv.presentation.databinding.ActivityMosaicResultBinding
+
+class MosaicResultActivity :
+    BaseActivity<ActivityMosaicResultBinding>(R.layout.activity_mosaic_result) {
+
+    companion object {
+        const val INTENT_KEY = "image"
+        const val SHARE_TYPE = "image/jpg"
+    }
+
+    private val viewModel: MosaicResultViewModel by viewModels()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        (if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            intent.getParcelableExtra(INTENT_KEY, Uri::class.java)
+        } else {
+            intent.getParcelableExtra(INTENT_KEY) as? Uri
+        })?.let { image ->
+            viewModel.setImage(
+                image
+            )
+            observeImage()
+        }
+
+        showSnackBarAction()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        initTopBar()
+        initClickListeners()
+    }
+
+    private fun initTopBar() {
+        binding.topBar.btClose.visible()
+        binding.topBar.ivShare.visible()
+        binding.topBar.tvSaveDone.visible()
+        binding.topBar.btBack.gone()
+        binding.topBar.tvTitle.gone()
+        binding.topBar.btNext.gone()
+        binding.topBar.btDownload.gone()
+    }
+
+    private fun observeImage() {
+        viewModel.image.observe(this) { image ->
+            binding.ivMosaicResult.setImageURI(image)
+        }
+    }
+
+    private fun initClickListeners() {
+        binding.topBar.ivShare.setOnClickListener {
+            val shareIntent: Intent = Intent().apply {
+                action = Intent.ACTION_SEND
+                putExtra(Intent.EXTRA_STREAM, viewModel.image.value)
+                type = SHARE_TYPE
+            }
+
+            startActivity(Intent.createChooser(shareIntent, R.string.share_text.toString()))
```

Comment:
```suggestion
            startActivity(Intent.createChooser(shareIntent, getString(R.string.share_text))
```
- `String 리소스`를 가져올때는 `Context` 함수인 `getString()`을 사용하시면 됩니다! 아마 현재 코드에서의 결과값은 `share_text`의 리소스 ID의 정수형을 나타내고 있을것으로 보입니다..

Reply:
수정했습니다~